app/services/bulk_update/add_trainees/import_row.rb

```diff
@@ -35,6 +35,7 @@ def call
         )
 
         trainee_attributes = trainee_attributes_service.new(mapper_klass.call(params: attributes))
+        trainee_attributes.record_source = :csv
```

Comment:
```suggestion
        trainee_attributes.record_source = :csv
```

I think `record_source` is already set at `prepare_record_source_attribute` but it is a `String`.